dll/win32/shell32/iconcache.cpp

```diff
@@ -881,13 +883,26 @@ UINT WINAPI ExtractIconExW(LPCWSTR lpszFile, INT nIconIndex, HICON * phiconLarge
         PrivateExtractIconExW = (UINT(WINAPI*)(LPCWSTR,int,HICON*,HICON*,UINT)) GetProcAddress(hUser32, "PrivateExtractIconExW");
 
         if (!PrivateExtractIconExW)
-        return 0;
+        return ret;
     }
 #endif
 
     TRACE("%s %i %p %p %i\n", debugstr_w(lpszFile), nIconIndex, phiconLarge, phiconSmall, nIcons);
+    ret = PrivateExtractIconExW(lpszFile, nIconIndex, phiconLarge, phiconSmall, nIcons);
+
+    /*
+     * PrivateExtractIconEx and the internal routine, ICO_ExtractIconEx, return -1 (the unsigned integer being 0xFFFFFFFF)
+     * in case the file is not a valid one (PE file image not being valid) or the file has not been found. ExtractIconExW()
+     * only returns the successful number of extracted icons from a file regardless if the file is not valid. In such
+     * scenario, just return 0.
+    */
+    if (ret == 0xFFFFFFFF)
+    {
+        WARN("ExtractIconExW(): An invalid file has been detected or the file couldn't be found! Failed to extract any icons!\n");
```

Comment:
Printing `lpszFile` here would be good (using dbgstr_w). You can shorten the message to make up for it (it can be assumed whoever wants to know more about this will read the source, so the message doesn't need to be very detailed).